### ISSUE TITLE
UI stuck in "Provisioning app backend", "Deleting backed env" and "Creating backed env" loop

### ISSUE THREAD

**Issue by ritesh-guaubox** [aws-amplify/amplify-console#1946]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v15.14.0

### Amplify CLI Version

4.52.0

### What operating system are you using?

Mac

### Amplify Categories

api

### Amplify Commands

push

### Describe the bug

The only thing I did was change a type from String to an enum type, in one of the models. and did amplify push --y
did similar things so many times this week but never faced an issue. 

The cli gets stuck at :

```
✔ Generated GraphQL operations successfully and saved at src/graphql
⠼ Updating resources in the cloud. This may take a few minutes...
GraphQL schema compiled successfully.
```

upon going to the amplify console online, I see:

![image](https://user-images.githubusercontent.com/81090456/121175370-52888a00-c89e-11eb-83d4-a8786f262e8e.png)

on refresh:
![image](https://user-images.githubusercontent.com/81090456/121175462-67fdb400-c89e-11eb-9e14-f77e307686d4.png)

on refresh:
![image](https://user-images.githubusercontent.com/81090456/121175610-99767f80-c89e-11eb-93ab-6b8fea4edae8.png)

then it goes back to completed, and never finishes. 

The change is deployed to AppSync, since I see it when I run my mutation. 

everything is fine but the backend is stuck and not allowing any more changes since its stuck in some unknown state. 

If I use a separate terminal instance and run amplify push with a small change, it throws:

`Failed to create models in the cloud: ValidationException - An error occurred while processing your request:  ValidationException: There is already a pending job for the requested appId: dckq06nq2cy6m and backendEnvironmentName: dev, wait for it to complete before creating a new job.`
 

### Expected behavior

cli finishes and backed is in completed state. 

### Reproduction steps

unable to reproduce

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by brnmye** [comment]
Hello could you please provide the appId and region for this app giving you issues? Also for the amplify push step did you simply use amplify push --y?

**Comment by ritesh-guaubox** [comment]
Hello @brnmye Thanks for looking into this, here you go:

app id: dckq06nq2cy6m
region: ap-southeast-2

yes. I just did `amplify push --y`

FYI even the admin UI is showing deploying status:

![Screen Shot 2021-06-15 at 6 37 43 pm](https://user-images.githubusercontent.com/81090456/122021418-ff1bbc00-ce08-11eb-87af-b0439d5ff09c.png)

**Comment by ritesh-guaubox** [comment]
Hello @brnmye, any update on this one?

**Comment by ritesh-guaubox** [comment]
Hello Team. I really am struggling to get this fixed. Its been 11 days since I opened the issue. and I have received no help. 
I have kept the issue on so that you can investigate and get back. Kept the env as is so that it can help you debug the issue. This is really concerning. 

FYI my backend env is still in the same limbo state.

My next step would be to delete it altogether and recreate it. In which case you might lose the logs and have some one else get this issue and repeat the entire thing allover. 

cc @brnmye

**Comment by ritesh-guaubox** [comment]
I just changed the schema and now I get this:
`Successfully generated models. Generated models can be found in --/backend-amplify/src
Failed to create models in the cloud: ValidationException - An error occurred while processing your request:  ValidationException: There is already a pending job for the requested appId: dckq06nq2cy6m and backendEnvironmentName: dev, wait for it to complete before creating a new job.`

**Comment by ritesh-guaubox** [comment]
Not sure if anyone is still looking at this but I am still getting:
"Successfully generated models. Generated models can be found in ~/gb-backend-amplify/src
Failed to create models in the cloud: ValidationException - An error occurred while processing your request:  ValidationException: There is already a pending job for the requested appId: dckq06nq2cy6m and backendEnvironmentName: dev, wait for it to complete before creating a new job."

its been 24 days. and there is still something running which is causing my push jobs to fail.

**Comment by akilisosa** [comment]
I am stuck in a loop as well. My app has been saying deletion in progress for a while. two days.

After not being able to update my lambda functions, or push new code, or whatever it was i was suppose to do with the command line because i lost support i thought. Geeze nobody really uses my app so im just going to delete this, and restart. but i have cannot delete it because it is still there in the deleting app space. 

![image](https://user-images.githubusercontent.com/18223349/124498336-29b6c000-dd8a-11eb-918b-94a662caaac6.png)

**Comment by ritesh-guaubox** [comment]
Yes. this loop is not going anywhere. I still have the issue. and the Amplify team hasn't cared to respond or look into it since a month. We have moved to a different tech stack due to this and many other issues which we can't risk in Production. This is frustrating. Welcome.

**Comment by guttyjr** [comment]
Hello,

I have same issue today.
I cannot resolve this issue. please tell me what how to recovery it.

![2022-01-10_12h13_59](https://user-images.githubusercontent.com/16617154/148714416-b40a1d58-8d8b-443e-8ccc-3a0110ae42fd.png)
